### PR TITLE
BUG: Fix return for RegressionResults

### LINKED ISSUE #4618: RegressionResults.__str__ returns None
Also it doesn't get called when accessed on the RegressionResultsWrapper object, since the wrapper has its own `__str__` method.

### PR DESCRIPTION
**Pull request by bashtage**
RR str was missing the return keyword

closes #4618

- [X] closes #4618
- [X] code/documentation is well formatted.  
- [X] properly formatted commit message. See 
      [NumPy's guide](https://docs.scipy.org/doc/numpy/dev/gitwash/development_workflow.html#writing-the-commit-message). 

<details>

**Notes**:

* It is essential that you add a test when making code changes. Tests are not 
  needed for doc changes.
* When adding a new function, test values should usually be verified in another package (e.g., R/SAS/Stata).
* When fixing a bug, you must add a test that would produce the bug in master and
  then show that it is fixed with the new code.
* New code additions must be well formatted. Changes should pass flake8. If on Linux or OSX, you can
  verify you changes are well formatted by running 
  ```
  git diff upstream/master -u -- "*.py" | flake8 --diff --isolated
  ```
  assuming `flake8` is installed. This command is also available on Windows 
  using the Windows System for Linux once `flake8` is installed in the 
  local Linux environment. While passing this test is not required, it is good practice and it help 
  improve code quality in `statsmodels`.
* Docstring additions must render correctly, including escapes and LaTeX.

</details>

### CODE REVIEW

File: statsmodels/regression/linear_model.py [hunk 1]
```diff
@@ -1522,7 +1522,7 @@ def __init__(self, model, params, normalized_cov_params=None, scale=1.,
             setattr(self, key, kwargs[key])
 
     def __str__(self):
-        self.summary()
+        return self.summary()
```
Comment (jbrockmendel) [inline statsmodels/regression/linear_model.py]: should be `str(self.summary())`

**Comment by jbrockmendel** [comment]
Needs test.  IIRC from when I tried to fix this, there was an issue with RegressionResultsWrapper returning not passing through `self._results.__str__` but using its own, which inherited from object

**Comment by bashtage** [comment]
> Needs test. IIRC from when I tried to fix this, there was an issue with RegressionResultsWrapper returning not passing through `self._results.__str__` but using its own, which inherited from object

That's ok.  I'm not trying to change the functionality, only the remove an obvious bug in the simplest way.

**Comment by jbrockmendel** [comment]
> That's ok. I'm not trying to change the functionality, only the remove an obvious bug in the simplest way.

I'm not sure I follow.  This seems like a "worth doing \Rightarrow worth doing right" situation

**Comment by bashtage** [comment]
Better to remove it since this pattern isn't used.

**Comment by jbrockmendel** [comment]
lgtm

**Pull request merged**